src/pipelines/similarity_baseline_json.py

```diff
@@ -0,0 +1,131 @@
+
+def filterText(doc, nlp):
+   
+    '''
+    Purpose - Removes stop words, out of vocabulary words and non-alpha words from a spacy document's vocabulary
+    Input - A spacy document and the spacy English parser
+    Output - A new spacy document
+    '''
+   
+    # Remove stop words, out of vocabulary and non-alpha tokens from document vocabulary   
+    newVocab = []
+    for token in doc:
+       if not token.is_stop and not token.is_oov and token.is_alpha: newVocab.append(token.orth_)
+    filteredDoc = nlp(' '.join(newVocab))
+
+    return filteredDoc 
+
+def assessSimilarity(allClusters, lookupOrder, documentData, nlp):
+
+    '''
+    Purpose - Do vector comparison and bag of words cosine similarity for each cluster found in JSON file
+    Input - A set of cluster IDs, a dict of document arrival order, an array of the parsed JSON document data, and a spacy English parser
+    Output - Array of namedtuples containing cluster_id, post_id, novelty, vector score, cosine similarity score
+    '''
+    
+    # Store results of similarity assessments
+    postScores = []
+    postScore = namedtuple('postScore','cluster_id,post_id,novelty,vectorScore,bagwordsScore')    
+
+    # Iterate through clusters found in JSON file, do similarity assessments, build a rolling corpus from ordered documents for each cluster
+    for cluster in allClusters:
+                
+        # Set corpus to first doc in this cluster and prepare to update corpus with new document vocabulary
+        corpus = filterText( nlp( documentData[ lookupOrder[ str(cluster)+"_0"] ] ["body_text"] ), nlp) 
+        updateCorpus = []
+        for token in corpus: updateCorpus.append(token.orth_)
+
+        i=1
+        search = True
+        while search is True:
+            if lookupOrder.has_key(str(cluster) + "_" + str(i)):
+                
+                # Find next document in order
+                index =  lookupOrder[str(cluster) + "_" + str(i)]
+                postID = documentData[index]["post_id"]
+                novelty = documentData[index]["novelty"]
+                doc = filterText(nlp(documentData[index]["body_text"]), nlp)
+
+                # Document vs Corpus vector comparison
+                vectorScore = doc.similarity(corpus) 
+
+                # Build Bag of Words with Spacy
+                docBagWords = doc.count_by(LOWER)
+                corpusBagWords = corpus.count_by(LOWER)
+                
+                # Combine Bag of Words dicts in vector format, calculate cosine similarity of resulting vectors  
+                vect = DictVectorizer(sparse=False)
+                bagwordsVectors = vect.fit_transform([docBagWords, corpusBagWords])
+                similarityScore = 1 - spatial.distance.cosine(bagwordsVectors[0], bagwordsVectors[1])
+                
+                # Save results in namedtuple and add to array
+                postScore = (cluster, postID, novelty, vectorScore, similarityScore)
```

Comment:
You reinstantiate postScore as a regular tuple here. postScore from above is a subclass (One place where python case convention might help. You could rename postScore above as PostScore and change this to `post_score = PostScore(...)`